tests/page-object-models/onboarding.page.ts

```diff
@@ -0,0 +1,75 @@
+import { Page } from '@playwright/test';
+import { TEST_PASSWORD, TEST_SECRET_KEY } from '@tests/constants';
+import { HomePageSelectors } from '@tests/selectors/home.selectors';
+import { OnboardingSelectors } from '@tests/selectors/onboarding.selectors';
+import { json } from '@tests/utils';
+
+import { RouteUrls } from '@shared/route-urls';
+
+const selectors = {};
+
+export class OnboardingPage {
+  readonly page: Page;
+  readonly selectors = selectors;
+
+  constructor(page: Page) {
+    this.page = page;
+  }
+
+  async select(selector: keyof typeof selectors) {
+    return this.page.$(selectors[selector]);
+  }
+
+  getSelector(selector: keyof typeof selectors) {
+    return this.selectors[selector];
+  }
+
+  async setupExtensionWithApiCalls(extensionId: string) {
+    await this.page.route(/.*/, route => route.continue());
+    await this.page.route(/chrome-extension/, route => route.continue());
+    await this.page.route(/github/, route => route.fulfill(json({})));
+    await this.page.goto(`chrome-extension://${extensionId}/index.html`);
+  }
+
+  async setupExtensionWithMockApiCalls(extensionId: string) {
+    await this.page.route(/.*/, route => route.abort());
+    await this.page.route(/chrome-extension/, route => route.continue());
+    await this.page.route(/github/, route => route.fulfill(json({})));
+    await this.page.goto(`chrome-extension://${extensionId}/index.html`);
+  }
```

Comment:
this aren't specific to onboarding, maybe can be declared outside of a page object. Separate mocks file?